libarchive/archive_read_support_format_mtree.c

```diff
@@ -1073,6 +1073,7 @@ read_mtree(struct archive_read *a, struct mtree *mtree)
 		/* Non-printable characters are not allowed */
 		for (s = p;s < p + len - 1; s++) {
 			if (!isprint((unsigned char)*s) && *s != '\t') {
+				fprintf(stderr, "non-printing character 0x%02X\n", (unsigned char)(*s));
```

Comment:
Please don't write to stderr from within libarchive.  This should be formatted as an error string `using archive_set_error()` for the client to display as they see fit.